core-management/core-management-subsystem/src/main/java/org/wildfly/extension/core/management/CoreManagementSubsystemSchema_1_0.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright The WildFly Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.wildfly.extension.core.management;
+
+import org.jboss.as.controller.Feature;
+import org.jboss.as.controller.PersistentResourceXMLDescription;
+import org.jboss.as.controller.PersistentResourceXMLDescription.PersistentResourceXMLBuilder;
+import org.jboss.as.controller.PersistentSubsystemSchema;
+import org.jboss.as.controller.SubsystemSchema;
+import org.jboss.as.controller.xml.VersionedNamespace;
+import org.jboss.as.version.Stability;
+import org.jboss.staxmapper.IntVersion;
+
+import java.util.EnumSet;
+import java.util.Map;
+
+import static org.jboss.as.controller.PersistentResourceXMLDescription.builder;
+
+/**
+ * Parser and Marshaller for core-management subsystem.
+ *
+ * <em>All resources and attributes must be listed explicitly and not through any collections.</em>
+ * This ensures that if the resource definitions change in later version (e.g. a new attribute is added),
+ * this will have no impact on parsing this specific version of the subsystem.
+ *
+ * @author <a href="http://jmesnil.net/">Jeff Mesnil</a> (c) 2016 Red Hat inc.
+ */
+public enum CoreManagementSubsystemSchema_1_0 implements PersistentSubsystemSchema<CoreManagementSubsystemSchema_1_0> {
+
+    VERSION_1_0(1),
+    VERSION_1_0_PREVIEW(1, Stability.PREVIEW);
+    static final Map<Stability, CoreManagementSubsystemSchema_1_0> ALL = Feature.map(EnumSet.of(VERSION_1_0, VERSION_1_0_PREVIEW));
+
+    private final VersionedNamespace<IntVersion, CoreManagementSubsystemSchema_1_0> namespace;
+
+    CoreManagementSubsystemSchema_1_0(int major) {
+        this.namespace = SubsystemSchema.createLegacySubsystemURN(CoreManagementExtension.SUBSYSTEM_NAME, new IntVersion(major, 0));
+    }
+
+    CoreManagementSubsystemSchema_1_0(int major, Stability stability) {
+        this.namespace = SubsystemSchema.createLegacySubsystemURN(CoreManagementExtension.SUBSYSTEM_NAME, stability, new IntVersion(major, 0));
+    }
+
+    @Override
+    public VersionedNamespace<IntVersion, CoreManagementSubsystemSchema_1_0> getNamespace() {
+        return this.namespace;
+    }
+
+    @Override
+    public PersistentResourceXMLDescription getXMLDescription() {
+        PersistentResourceXMLBuilder builder =  builder(CoreManagementExtension.SUBSYSTEM_PATH, namespace);
```

Comment:
I recommend that you create a PersistentResourceXMLDescription.Factory for creating PersistentResourceXMLBuilder instances (instead of using the static builder(...) methods.  That way you don't need to do any stability-level filtering when creating your PersistentResourceXMLDescription.
Come to think of it, I should probably deprecate the existing static build(...) methods to discourage their use.
e.g.
```java
PersistentResourceXMLDescription.Factory factory = PersistentResourceXMLDescription.factory(this);
PersistentResourceXMLBuilder builder =  factory.builder(CoreManagementExtension.SUBSYSTEM_PATH);
// ...
factory.builder(UnstableApiAnnotationResourceDefinition.RESOURCE_REGISTRATION).addAttributes(...);
```

Reply:
ah ok, I simply copied the old stuff and adjusted :-)